packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -347,37 +361,42 @@ export class ComputeEnvironment extends Resource implements IComputeEnvironment
 
     // Only allow compute resources to be set when using MANAGED type
     if (props.computeResources && this.isManaged(props)) {
+      const isFargate = ComputeResourceType.FARGATE === props.computeResources.type
+        || ComputeResourceType.FARGATE_SPOT === props.computeResources.type;
+
       computeResources = {
-        allocationStrategy: props.computeResources.allocationStrategy
-         || (
-           props.computeResources.type === ComputeResourceType.SPOT
-             ? AllocationStrategy.SPOT_CAPACITY_OPTIMIZED
-             : AllocationStrategy.BEST_FIT
-         ),
         bidPercentage: props.computeResources.bidPercentage,
         desiredvCpus: props.computeResources.desiredvCpus,
         ec2KeyPair: props.computeResources.ec2KeyPair,
         imageId: props.computeResources.image && props.computeResources.image.getImage(this).imageId,
-        instanceRole: props.computeResources.instanceRole
-          ? props.computeResources.instanceRole
-          : new iam.CfnInstanceProfile(this, 'Instance-Profile', {
-            roles: [new iam.Role(this, 'Ecs-Instance-Role', {
-              assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
-              managedPolicies: [
-                iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonEC2ContainerServiceforEC2Role'),
-              ],
-            }).roleName],
-          }).attrArn,
-        instanceTypes: this.buildInstanceTypes(props.computeResources.instanceTypes),
         launchTemplate: props.computeResources.launchTemplate,
         maxvCpus: props.computeResources.maxvCpus || 256,
-        minvCpus: props.computeResources.minvCpus || 0,
         placementGroup: props.computeResources.placementGroup,
         securityGroupIds: this.buildSecurityGroupIds(props.computeResources.vpc, props.computeResources.securityGroups),
         spotIamFleetRole: spotFleetRole?.roleArn,
         subnets: props.computeResources.vpc.selectSubnets(props.computeResources.vpcSubnets).subnetIds,
         tags: props.computeResources.computeResourcesTags,
         type: props.computeResources.type || ComputeResourceType.ON_DEMAND,
+        ...(!isFargate ? {
```

Comment:
I wonder - why are we implicitly ignoring `allocationStrategy`, `instanceRole`, `instanceTypes`, and `minvCpus` instead of throwing an exception if they are configured alongside a fargate environment? 

There are plenty of other properties we disallow specifying for fargate, so why not treat these the same? 

Feels like if the user explicitly set these properties we should throw an exception, and the user didn't - then we should apply defaults only if its a non fargate environment.

Reply:
Agreed, updated

Reply:
Agreed, updated